server/svix-server/src/cfg.rs

```diff
@@ -139,6 +147,17 @@ pub struct ConfigurationInner {
     #[validate(range(min = 1, max = 30))]
     pub worker_request_timeout: u16,
 
+    /// How long of a period to wait after the first dispatch failure to disable an endpoint. If a
+    /// message is successfully sent during this time, then the endpoint will not disable.
+    #[serde(deserialize_with = "deserialize_hours")]
+    pub dispatch_disable_grace_period: Duration,
+
+    /// How long of a period to wait after a first failure has been cached to remove the record. If
+    /// a consistently failing endpoint is called only once within this period, then it will never
+    /// be disabled.
+    #[serde(deserialize_with = "deserialize_hours")]
+    pub dispatch_disable_expiration_period: Duration,
```

Comment:
You need a new config value: `disable_endpoint_in` which disables after X hours.

This shouldn't be a config, just set it to 60-80% of the new one we'll add.